Octavia Ingress Controller Fails with latest

<!-- This form is for bug reports and feature requests! -->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature


**What happened**:
So a few months ago I started toying with the openstack provider, specifically the octavia ingress stuff as it would have been useful to us. I ended up having to drop it as our provider was on version Pike, which didn’t support bulk creation of members for LB’s. Well they updated the version to Queens which has that functionality, so I went back to it. After fixing an issue I had with auth, the statefulset made it past that, however, the pod for the ingress controller panics and crashes using the latest tag from docker hub.

**What you expected to happen**:
Expect stateful set to start and controller to be available.

**How to reproduce it (as minimally and precisely as possible)**:
Simply follow the instructions in https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-octavia-ingress-controller.md, using the latest tag for the image.

**Anything else we need to know?**:
Looked at the page for the image in Docker Hub and latest was updated 9 hours ago. So I changed the tag from latest to v1.15.0 and the controller comes online. The errors from the logs are as so:

`time="2019-09-10T14:47:58Z" level=info msg="Using config file" file=/etc/config/octavia-ingress-controller-config.yaml
W0910 14:47:58.803504 1 client_config.go:541] Neither --kubeconfig nor --master was specified. Using the inClusterConfig. This might not work.
E0910 14:47:59.414664 1 runtime.go:69] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x13a7360), concrete:(*runtime._type)(0x154c0a0), asserted:(*runtime._type)(0x154d720), missingMethod:""} (interface conversion: interface {} is *v1beta1.Ingress, not *v1beta1.Ingress (types from different packages))
goroutine 55 [running]:
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x13deec0, 0xc0006bb6e0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65 +0x7b
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:47 +0x82
panic(0x13deec0, 0xc0006bb6e0)
/usr/local/go/src/runtime/panic.go:522 +0x1b5
k8s.io/cloud-provider-openstack/pkg/ingress/controller.NewController.func1(0x154c0a0, 0xc000279c80)
/home/zuul/src/k8s.io/cloud-provider-openstack/pkg/ingress/controller/controller.go:242 +0x3bc
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/controller.go:196
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0xc000278a80, 0xc0002b64b0, 0x40b6a9)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/shared_informer.go:642 +0x26d
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc00048fe38, 0x42ac1f, 0xc0002b64e0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:284 +0x51
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/shared_informer.go:636 +0x79
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0006adf68)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00048ff68, 0xdf8475800, 0x0, 0x42b401, 0xc000086de0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc0002bd980)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/shared_informer.go:634 +0x9c
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0005a61b0, 0xc000360940)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x4f
created by k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62
panic: interface conversion: interface {} is *v1beta1.Ingress, not *v1beta1.Ingress (types from different packages) [recovered]
panic: interface conversion: interface {} is *v1beta1.Ingress, not *v1beta1.Ingress (types from different packages)
goroutine 55 [running]:
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:54 +0x105
panic(0x13deec0, 0xc0006bb6e0)
/usr/local/go/src/runtime/panic.go:522 +0x1b5
k8s.io/cloud-provider-openstack/pkg/ingress/controller.NewController.func1(0x154c0a0, 0xc000279c80)
/home/zuul/src/k8s.io/cloud-provider-openstack/pkg/ingress/controller/controller.go:242 +0x3bc
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/controller.go:196
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0xc000278a80, 0xc0002b64b0, 0x40b6a9)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/shared_informer.go:642 +0x26d
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc00006be38, 0x42ac1f, 0xc0002b64e0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:284 +0x51
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/shared_informer.go:636 +0x79
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0006adf68)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00006bf68, 0xdf8475800, 0x0, 0x42b401, 0xc000086de0)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc0002bd980)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/client-go/tools/cache/shared_informer.go:634 +0x9c
k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0005a61b0, 0xc000360940)
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x4f
created by k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
/home/zuul/src/k8s.io/cloud-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62`

**Environment**:
- openstack-cloud-controller-manager version:
- OS (e.g. from /etc/os-release): ubuntu 18.04

@oomichi Thanks, not sure what i'm looking for after all.. This might be real long shot but in the stacktrace there's these two lines that caught my eye:
```
panic: interface conversion: interface {} is *v1beta1.Ingress, not *v1beta1.Ingress (types from different packages)
```
and
```
/home/zuul/src/k8s.io/cloud-provider-openstack/pkg/ingress/controller/controller.go:242 +0x3bc
```
In `controller.go` on line `242` the following happens:
https://github.com/kubernetes/cloud-provider-openstack/blob/23ac807a68fbf3eece678bb5977dcf64559568a9/pkg/ingress/controller/controller.go#L242
This was changed in PR [658](https://github.com/kubernetes/cloud-provider-openstack/pull/658) from `extensions/v1beta1` to `networking.k8s.io/v1beta1` since the former one will be deprecated in v1.18.

When searching for more info i found this issue: https://github.com/golang/go/issues/26094. As i said, this is a long shot, but could there be something in the issue above (and the fact that the API of Ingress was changed) that crashes the Ingress controller? Or could it be a combination of versions of different imported packages causing this?

Ping @lingxiankong, does this make any sense to you?

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -1794,7 +1794,7 @@
     "k8s.io/client-go/kubernetes/scheme",
     "k8s.io/client-go/kubernetes/typed/core/v1",
     "k8s.io/client-go/listers/core/v1",
-    "k8s.io/client-go/listers/extensions/v1beta1",
+    "k8s.io/client-go/listers/networking/v1beta1",
     "k8s.io/client-go/rest",
     "k8s.io/client-go/tools/cache",
     "k8s.io/client-go/tools/clientcmd",

--- a/pkg/ingress/controller/controller.go
+++ b/pkg/ingress/controller/controller.go
@@ -36,7 +36,7 @@ import (
 	"k8s.io/client-go/kubernetes/scheme"
 	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
 	corelisters "k8s.io/client-go/listers/core/v1"
-	extlisters "k8s.io/client-go/listers/extensions/v1beta1"
+	extlisters "k8s.io/client-go/listers/networking/v1beta1"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/record"
@@ -236,7 +236,7 @@ func NewController(conf config.Config) *Controller {
 		kubeClient:          kubeClient,
 	}
 
-	ingInformer := kubeInformerFactory.Extensions().V1beta1().Ingresses()
+	ingInformer := kubeInformerFactory.Networking().V1beta1().Ingresses()
 	ingInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			addIng := obj.(*nwv1beta1.Ingress)

